Mathlib/Analysis/Normed/Group/Basic.lean

```diff
@@ -490,21 +490,25 @@ theorem norm_nonneg' (a : E) : 0 ≤ ‖a‖ := by
 #align norm_nonneg' norm_nonneg'
 #align norm_nonneg norm_nonneg
 
-/- porting note: meta code, do not port
-section
+namespace Mathlib.Meta.Positivity
 
-open Tactic Tactic.Positivity
+open Lean Meta Qq Function
 
 /-- Extension for the `positivity` tactic: norms are nonnegative. -/
-@[positivity]
-unsafe def _root_.tactic.positivity_norm : expr → tactic strictness
-  | q(‖$(a)‖) =>
-    nonnegative <$> mk_app `` norm_nonneg [a] <|> nonnegative <$> mk_app `` norm_nonneg' [a]
-  | _ => failed
-#align tactic.positivity_norm tactic.positivity_norm
+@[positivity Norm.norm _]
+def evalMulNorm : PositivityExt where eval {_ _} _zα _pα e := do
+  let .app _ a ← withReducible (whnf e) | throwError "not ‖ ⬝ ‖"
+  let p ← mkAppM ``norm_nonneg' #[a]
+  pure (.nonnegative p)
 
-end
--/
+/-- Extension for the `positivity` tactic: norms are nonnegative. -/
```

Comment:
Would be good if the comment indicated that this was just the *additive* case. Maybe:
```lean
/-- Extension for the `positivity` tactic: additive norms are nonnegative, via `norm_nonneg`. -/
```

Reply:
```suggestion
/-- Extension for the `positivity` tactic: additive norms are nonnegative, via `norm_nonneg`. -/
```